### PR TITLE
WIP: Add !alias command

### PR DESCRIPTION
**Pull request by zpatrick**
Adds a `!alias` command and behavior to the slackbot. 

### TODO
- [x] `!alias show NAME` - shows the pattern and template for the specified alias
- [x] `!alias rm NAME`
- [x] `!alias test NAME TEXT` - tests the alias with the specified name against the specified text. Displays the transformed text back to the user. 
- [x] Trigger a cache invalidation when an alias is added or removed
- [x] Unit Tests

### CODE REVIEW

File: bot/ping_command.go [hunk 1]
```diff
@@ -1,23 +0,0 @@
-package bot
```
Comment (zpatrick) [inline bot/ping_command.go]: Figured I could try dog-fooding this by using `!alias ping "!ping" "!echo pong"`

File: main.go [hunk 2]
```diff
@@ -136,6 +136,7 @@ func main() {
 
 	slackbot.Action = func(c *cli.Context) error {
 		behaviors := bot.Behaviors{
+			bot.NewAliasBehavior(store).Behavior(),
```
Comment (zpatrick) [inline main.go]: Should I follow `<entity><verb>Behavior` naming like `KarmaTrackingBehavior`, e.g. `AliasTransformingBehavior`?

**Comment by tlake** [comment]
For visibility, I'm gonna summarize some conversation about `!alias` I had with @zpatrick in a PM.

# Dangerous Glob

Given a usage like `!alias add "PATTERN" "TEMPLATE"`, I would caution very strongly against glob matching on `PATTERN`. That creates a pretty big security (is that the right term?) breach, in that someone could write this alias:
```
!alias add "*" "!gif something nsfw"
```
and then _every single message posted across every channel in the slack team gets matched to that alias and replaced with an nswf gif._ Even `!alias` commands like `!alias rm` would match that, so you couldn't even find the offending alias and remove it.

One suggestion was to skip glob matching if the message received starts with `!alias`. That way, you could never overwrite the `!alias` command with an alias. Even then, although you could find and remove bogus aliases, all of slack is still nswf gifs until someone takes the time to fix it.

# Issues with `.Text`

We could try and keep `!alias` super simple and do no glob matching at all, which is certainly safe, but very, very limiting. It could be just a simple substitution feature that passes through the rest of user input:
```
!alias add "!wfh *" "<@{{ .User }}> is working from home today! {{ .Text }}"
```
but that would probably require a bit of backend work on `.Text`, since at the moment it's the full message text and that would include the `!alias` as well:
```
!wfh cc person1 person2

output:
<@tlake> is working from home today! !wfh cc person1 person2
```
Maybe something like `name, input := strings.SplitN(msg.Text, " ", 2)` somewhere on the backend?

**Comment by zpatrick** [comment]
merging for now, can always come back and re-address

**Pull request merged**